rclpy/rclpy/node.py

```diff
@@ -402,6 +404,11 @@ def get_parameter(self, name: str) -> Parameter:
         """
         Get a parameter by name.
 
+        If the parameter was not declared, if undeclared parameters are allowed and the parameter
+        is one of the initial parameters, it will be implicitly declared with a default descriptor
+        and returned.
```

Comment:
I don't think it should be implicitly declared when getting it, the `rclcpp` version will just return a default initialized `Parameter` (type `PARAMETER_NOT_SET`) and not touch the internal node's state at all.

Imo, only `set_*` should implicitly declare something when `allow_undeclared_parameters` is used. We don't have a test case for this in `rclcpp`, but the documentation doesn't mention that it will be implicitly declared:

https://github.com/ros2/rclcpp/blob/d399fef9c6f90462c15b57d9871a5e3c7ceda548/rclcpp/include/rclcpp/node.hpp#L621-L639

See also the implementation:

https://github.com/ros2/rclcpp/blob/d399fef9c6f90462c15b57d9871a5e3c7ceda548/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp#L555-L567

Reply:
Alright, got it! Thanks for the references!

Reply:
Addressed in https://github.com/ros2/rclpy/pull/339/commits/9227ad2e77040f1da47f431d0695f72dab84ec6b; waiting for CI now.